### PR TITLE
Update to gcc 13.2.0 -- Needed for Ubuntu 23.10

### PR DESCRIPTION
**Pull request by hmaarrfk**
See https://gcc.gnu.org/gcc-13/

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
I feel like I'm hitting a repeat of https://github.com/conda-forge/ctng-compilers-feedstock/issues/95  for ubuntu 23.10

**Comment by hmaarrfk** [comment]
Ummmm:

> For C++, construction of the global iostream objects std::cout, std::cin, etc. is now done inside the standard library, instead of in every source file that includes the <iostream> header. This change improves the start-up performance of C++ programs, but it means that code compiled with GCC 13.1 will crash if the correct version of libstdc++.so is not used at runtime. See the [documentation](https://gcc.gnu.org/onlinedocs/libstdc++/manual/using_dynamic_or_shared.html#manual.intro.using.linkage.dynamic) about using the right libstdc++.so at runtime. Future GCC releases will mitigate the problem so that the program cannot be run at all with an older libstdc++.so.

Really not sure what to do about this. @h-vetinari has anything been using GCC 13 for now?

File: .ci_support/linux_64_cross_target_platformlinux-s390xgcc_machines390xgcc_maj_ver13gcc_version11.4.0glibc_version2.17.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ docker_image:
 gcc_machine:
 - s390x
 gcc_maj_ver:
-- '11'
+- '13'
```
Comment (hmaarrfk) [inline .ci_support/linux_64_cross_target_platformlinux-s390xgcc_machines390xgcc_maj_ver13gcc_version11.4.0glibc_version2.17.yaml:18]: is this change safe?

**Comment by hmaarrfk** [comment]
I can confirm that this helps get thigns working for ubuntu 23.10

**Comment by hmaarrfk** [comment]
thank you!

**Comment by h-vetinari** [comment]
> This change improves the start-up performance of C++ programs, but it means that code compiled with GCC 13.1 will crash if the correct version of libstdc++.so is not used at runtime.

This is pretty wild IMO, but luckily it seems that [zero](https://github.com/search?q=org%3Aconda-forge+%2F%28%3Fm%29c_compiler_version%3A%5Cs%2B%5C%23%5Cs%5C%5B%28unix%7Clinux%29%5C%5D%5Cn%5Cs*-+13%2F&type=code) feedstocks are using GCC 13 so far.

**Comment by hmaarrfk** [comment]
it seems that this note was present even in GCC 13.1. I believe maybe it is warning users that they must use libstdc++ built with 13+ or else face serious consequences.

I think we do that at conda forge right?

**Comment by isuruf** [comment]
Shouldn't be a problem for us

**Comment by hmaarrfk** [comment]
Thanks for confirming!

**Pull request merged**